pkg/operatorsource/datastore.go

```diff
@@ -0,0 +1,54 @@
+package operatorsource
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/operator-framework/operator-marketplace/pkg/appregistry"
+)
+
+var (
+	ErrManifestNotFound = errors.New("manifest not found")
+)
+
+type DatastoreReader interface {
+	// Read returns the associated operator manifest given a package ID
+	Read(packageID string) (*appregistry.OperatorMetadata, error)
+}
+
+func newDatastore() *hashmapDatastore {
+	return &hashmapDatastore{
+		list: map[string]*appregistry.OperatorMetadata{},
+	}
+}
+
+type hashmapDatastore struct {
```

Comment:
Please change the name to something appropriate like DataStore

Reply:
Hmm, do you have any suggestions? `MemoryDatastore`, `TransientDatastore` or `DummyDatastore` maybe? :)